spring-web/src/main/java/org/springframework/web/cors/DefaultCorsProcessor.java

```diff
@@ -126,7 +126,23 @@ protected boolean handleInternal(ServerHttpRequest request, ServerHttpResponse r
 		List<String> requestHeaders = getHeadersToUse(request, preFlightRequest);
 		List<String> allowHeaders = checkHeaders(config, requestHeaders);
 
-		if (allowOrigin == null || allowMethods == null || (preFlightRequest && allowHeaders == null)) {
+		if (allowOrigin == null) {
+			logger.debug("rejecting request because CORS processor cannot determine " +
```

Comment:
What about a message like `"Rejecting CORS request because '" + requestOrigin + "' origin is not allowed"`